libavfilter/vf_libvmaf.c

```diff
@@ -46,6 +48,13 @@
 #include "libavutil/hwcontext_cuda_internal.h"
 #endif
 
+typedef struct FrameList {
+    AVFrame *frame;
+    unsigned frame_number;
+    unsigned propagated_handlers_cnt;
+    struct FrameList *next;
+} FrameList;
```

Comment:
I wonder if we could use something like av_fifo here instead?

https://ffmpeg.org/doxygen/trunk/group__lavu__fifo.html

Reply:
Oh yes! Thats what I was looking for!

Reply:
@kylophone I make some experiments with `av_fifo` and that might be not a good idea. Vmaf can send frames any order, If we use FIFO then we should make some operations to send them in order.